src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -1241,9 +1243,33 @@ class BookingService(
       createCas1CancellationDomainEvent(booking, user, cancelledAt, reason)
     }
 
+    updateApplicationStatusOnCancellation(
+      booking = booking,
+      isUserRequestedWithdrawal = withdrawalContext.triggeringEntityType == WithdrawableEntityType.Booking
+    )
+
     return success(cancellationEntity)
   }
 
+  private fun updateApplicationStatusOnCancellation(
+    booking: BookingEntity,
+    isUserRequestedWithdrawal: Boolean
+  ) {
+    if(!isUserRequestedWithdrawal || booking.application == null) {
+      return
+    }
+
+    val application = booking.application!!
+    val bookings = bookingRepository.findAllByApplication(application)
+    val anyActiveBookings = bookings.any { it.isActive() }
+    if(!anyActiveBookings) {
+      applicationService.updateApprovedPremisesApplicationStatus(
+        application.id,
+        ApprovedPremisesApplicationStatus.AWAITING_PLACEMENT
```

Comment:
in the commit msg you talk about `ApprovedPremisesApplicationStatus.PLACEMENT_REQUIRED` which I think is imaginary...

Reply:
Good spot, that's now fixed